### ISSUE TITLE
setCustomValidityValueMissing value not showing when x icon button is clicked

### ISSUE THREAD

**Issue by cynthiaricomendoza-alaskaair** [AlaskaAirlines/auro-datepicker#176]
## Describe the bug
we've upgraded to the latest version on the auro-datepicker because we noticed the setCustomValidityRangeUnderflow property wasn't working as expected in the version we were using before. however, now we are seeing an issue with setCustomValidityValueMissing, but only when the X icon button is clicked on. when user selects a date then clears it by clicking the x icon button in the datepicker the help text is not shown. 

here is an example of what our code looks like:
```html
        <auro-datepicker
          id="birth-date"
          setCustomValidityValueMissing="Infant's birthdate is required."
          ${ref(this.liBirthdayRef)}
          @auroDatePicker-valueSet=${this.birthdaySelected}
          maxDate="12/13/2023"
          minDate="12/14/2021"
          setCustomValidityRangeUnderflow="Each child 2 years and older must have their own seat and ticket."
          required
        >
          <span slot="fromLabel"
            >Birth date</span
          >
          <span slot="mobileDateLabel"
            >Birth date</span
          >
        </auro-datepicker>
```

## To Reproduce

1. add a value for setCustomValidityValueMissing in the auro-datepicker
2. click on the auro-datepicker and input a date
3. click on the x icon button on the auro-datepicker
4. observe the value you set for setCustomValidityValueMissing is not shown under the input box in the datepicker

## Expected behavior

the setCustomValidityValueMissing value should be shown when the user clears the date value.

## Screenshots

https://github.com/AlaskaAirlines/auro-datepicker/assets/116121736/8fbb475f-6754-4a3b-b3ba-3b0a6e4aae0c

## Desktop (please complete the following information):

 - OS: macOS
 - Browser Chrome
 - Version 

## Additional context
This works as expected when a user goes in the input and deletes it in there.

**Comment by blackfalcon** [comment]
Hey @cynthiaricomendoza-alaskaair are you able to repro this on the Auro doc site? Also please validate that you are on the most current version. https://auro.alaskaair.com/components/auro/datepicker/api#required

Here is a screenshot of my repro of this issue. The behavior is as expected. 

<img width="616" alt="Screen Shot 2023-12-20 at 4 49 57 PM" src="https://github.com/AlaskaAirlines/auro-datepicker/assets/181089/ea6285ba-df7c-4d5f-bf9d-66898f04ec46">

**Comment by cynthiaricomendoza-alaskaair** [comment]
@blackfalcon no, i wasn't able to reproduce it on the auro doc site. in the screen recording i attached, i was using the latest version 2.0.10. this might be specific to the auro-input because i was doing some testing with the input alone and saw similar behavior: https://github.com/AlaskaAirlines/auro-input/issues/275

**Comment by blackfalcon** [comment]
@cynthiaricomendoza-alaskaair please try and schedule some time with either @jason-capsule42 or @jordanjones243 to review this issue. I am unable to repro your issue and we need to understand better the context of use.

**Comment by Patrick-Daly-AA** [comment]
validate in DatePicker Beta Branch
